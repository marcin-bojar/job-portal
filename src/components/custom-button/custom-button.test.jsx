import React from 'react';
import { shallow } from 'enzyme';

import CustomButton from './custom-button.component';

describe('CustomButton component', () => {
  it('renders CustomButton correctly', () => {
    const mockProps = {
      children: 'Zaloguj',
    };

    expect(shallow(<CustomButton {...mockProps} />)).toMatchSnapshot();
    expect(shallow(<CustomButton {...mockProps} />).props().children).toEqual(
      'Zaloguj'
    );
  });

  it('handles click event', () => {
    const mockCallback = jest.fn();
    const mockProps = {
      children: 'Zaloguj',
      onClick: mockCallback,
    };

    expect(mockCallback.mock.calls.length).toEqual(0);
    shallow(<CustomButton {...mockProps} />).simulate('click');
    expect(mockCallback.mock.calls.length).toEqual(1);
    shallow(<CustomButton {...mockProps} />).simulate('click');
    expect(mockCallback.mock.calls.length).toEqual(2);
  });

  it('renders Google sign in button correctly', () => {
    const mockProps = {
      isGoogleSignIn: true,
    };

    expect(
      shallow(<CustomButton {...mockProps} />).hasClass('google custom-button')
    );
  });
});
