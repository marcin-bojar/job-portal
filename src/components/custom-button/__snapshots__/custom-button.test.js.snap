// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton component renders CustomButton correctly 1`] = `
<button
  className=" custom-button"
>
  Zaloguj
</button>
`;
